crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -109,18 +110,65 @@ where
     ) -> Result<(), TransactionKernelError> {
         let pub_key = process.get_stack_word(0);
         let msg = process.get_stack_word(1);
+
         let signature_key = Hasher::merge(&[pub_key, msg]);
 
         let signature = if let Ok(signature) =
             process.advice_provider().get_mapped_values(&signature_key)
         {
             signature.to_vec()
         } else {
-            let signing_inputs = SigningInputs::Blind(msg);
+            // Retrieve transaction summary commitments from the advice provider.
+            // The commitments are stored as a contiguous array of field elements with the following
+            // layout:
+            // - commitments[0..4]:  SALT
+            // - commitments[4..8]:  OUTPUT_NOTES_COMMITMENT
+            // - commitments[8..12]: INPUT_NOTES_COMMITMENT
+            // - commitments[12..16]: ACCOUNT_DELTA_COMMITMENT
+            let commitments = process
+                .advice_provider()
+                .get_mapped_values(&msg)
+                .map_err(|err| TransactionKernelError::TransactionSummaryError(Box::new(err)))?;
+
+            if commitments.len() != 16 {
+                return Err(TransactionKernelError::TransactionSummaryError(
+                    "Expected 4 words for transaction summary commitments".into(),
+                ));
+            }
+
+            let extract_word = |start: usize| -> Word {
+                Word::from([
+                    commitments[start],
+                    commitments[start + 1],
+                    commitments[start + 2],
+                    commitments[start + 3],
+                ])
+            };
```

Comment:
nit: I would maybe define this as a small helper function at the end of the file.